packages/aws-cdk-lib/core/lib/cfn-resource.ts

```diff
@@ -725,7 +734,20 @@ function deepMerge(target: any, ...sources: any[]) {
         // sibling concrete values alongside, so we can delete this tree.
         const output = target[key];
         if (typeof(output) === 'object' && Object.keys(output).length === 0) {
-          delete target[key];
+          /*
+          * some empty trees are needed in some edge cases
+          * ie: wafv2 Rules (for Action, OverrideAction, etc...)
```

Comment:
Please clarify which `some empty trees` and maybe combine the `ie` into the `some edge cases` like `... are needed when WAFv2 rule action is one of 'Action', 'Block', 'Captcha', 'Challenge' or 'Count'.`